### PR TITLE
use context manager for snowflake connection in dask delayed functions

### PR DESCRIPTION
**Pull request by rikturr**
Also updated examples-gpu/nyc-taxi-snowflake/rf-rapids-dask.ipynb to use `fetch_pandas_all()`. We previously had to use `fetchall()` because of arrow version mismatches. That has been fixed so we are good to use `fetch_pandas_all()`.

### CODE REVIEW

File: examples/examples-cpu/nyc-taxi-snowflake/hyperparameter-dask.ipynb [hunk 1]
```diff
@@ -160,10 +160,10 @@
...
     "@dask.delayed\n",
     "def load(conn_info, query, day):\n",
-    "    conn = snowflake.connector.connect(**conn_info)\n",
-    "    taxi = conn.cursor().execute(query, str(day)).fetch_pandas_all()\n",
-    "    taxi.columns = taxi.columns.str.lower()\n",
-    "    return taxi    "
+    "    with snowflake.connector.connect(**conn_info) as conn:\n",
+    "        taxi = conn.cursor().execute(query, str(day)).fetch_pandas_all()\n",
+    "        taxi.columns = taxi.columns.str.lower()\n",
+    "        return taxi    "
```
Comment (jameslamb) [inline examples/examples-cpu/nyc-taxi-snowflake/hyperparameter-dask.ipynb]:
```suggestion
    "        return taxi"
```

File: examples/examples-cpu/nyc-taxi-snowflake/xgboost-dask.ipynb [hunk 2]
```diff
@@ -174,10 +174,10 @@
...
     "@dask.delayed\n",
     "def load(conn_info, query, day):\n",
-    "    conn = snowflake.connector.connect(**conn_info)\n",
-    "    taxi = conn.cursor().execute(query, str(day)).fetch_pandas_all()\n",
-    "    taxi.columns = taxi.columns.str.lower()\n",
-    "    return taxi    "
+    "    with snowflake.connector.connect(**conn_info) as conn:\n",
+    "        taxi = conn.cursor().execute(query, str(day)).fetch_pandas_all()\n",
+    "        taxi.columns = taxi.columns.str.lower()\n",
+    "        return taxi    "
```
Comment (jameslamb) [inline examples/examples-cpu/nyc-taxi-snowflake/xgboost-dask.ipynb]:
```suggestion
    "        return taxi"
```

**Review by jameslamb — CHANGES_REQUESTED** [review]
These changes look good to me! While you're touching this code, can you remove trailing whitespace as well by accepting the suggestions I left in your browser?

**Review by jameslamb — APPROVED** [review]
nice!

**Comment by hhuuggoo** [comment]
Task linked: [DATA-125 examples: Close connection in snowflake dask delayed functions](https://app.clickup.com/t/1250339/DATA-125)

**Pull request merged**